pallets/polkapool/src/lib.rs

```diff
@@ -327,11 +345,28 @@ decl_module! {
         pub fn slash_stake(origin, account: T::AccountId) -> DispatchResult {
             let origin = <T as Config>::Origin::from(origin);
             T::GovernanceOrigin::ensure_origin(origin)?;
-            let stake = <StakedUsers<T>>::get(&account);
-            <StakedUsers<T>>::remove(&account);
-            T::Currency::slash(AssetId::POLKADEX,&account,stake.staked_amount);
-            Self::deposit_event(RawEvent::StakeSlashed(account,stake.staked_amount));
+            let mut stake_info: StakeInfo<T> = <StakedUsers<T>>::take(&account);
+            T::Currency::withdraw(AssetId::POLKADEX,&account,stake_info.total_stake())?;
+            Self::deposit_event(RawEvent::StakeSlashed(account,stake_info.total_stake()));
             Ok(())
         }
     }
 }
+
+impl<T: Config> Module<T> {
+    // Calculates the stake amount and staking period
+    // The staking period will vary between 28-56 days. The heavier the transaction, longer the period.
+    // A transaction consuming the full allocated weight lock the tokens for 56 days.
+    pub fn calculate_stake_params(stake_price: T::Balance, call_weight: Weight) -> (T::Balance, T::BlockNumber) {
```

Comment:
Assume the fraction of total allocated block weight used by the call is `x`, then the staking period is set as  `MinStakePeriod*x + MinStakePeriod`. Therefore, if a call consumes the complete weight, then the staking period will be `2*MinStakePeriod`

Reply:
I am still researching new ways to calculate the stake amount and stake period based on the frequency of use, call weight, etc